### PR TITLE
feat: compressed account modifications

### PR DESCRIPTION
**Pull request by ananas-block**

### CODE REVIEW

File: cli/src/psp-utils/buildCircom.ts [hunk 1]
```diff
@@ -140,38 +146,15 @@ export async function generateCircuit({
...
-  await executeAnchor({ args: ["build"] });
-  console.log("✅ Build finished successfully");
+  fs.unlinkSync(path.join(sdkBuildCircuitDir, `${circuitName}Main.r1cs`));
+  fs.unlinkSync(path.join(sdkBuildCircuitDir, `${circuitName}Main.sym`));
 }
+function camelToKebab(str: string): string {
+  return str
+    .replace(/([a-z0-9]|(?=[A-Z]))([A-Z])/g, '$1-$2')  // Match camelCasing and insert "-"
+    .toLowerCase();  // Convert the entire string to lowercase
+}
```
Comment (sergeytimoshin) [inline cli/src/psp-utils/buildCircom.ts:160]: Shouldn't it be in the [cli/src/psp-utils/utils.ts, close to the "toSnakeCase" function](https://github.com/Lightprotocol/light-protocol/blob/main/cli/src/psp-utils/utils.ts#L39)?

Comment (ananas-block) [reply cli/src/psp-utils/buildCircom.ts:160]: good point, I think these functions are scattered, let me group these functions in zk.js.

File: cli/src/commands/psp/init.ts [hunk 2]
```diff
@@ -20,32 +12,22 @@ export default class InitCommand extends Command {
     }),
   };
 
+  static flags = {
+    circom: Flags.boolean({
+      description:
+        "Whether the main circuit is a circom circuit not a .light file.",
```
Comment (SwenSchaeferjohann) [inline cli/src/commands/psp/init.ts:18]:
```suggestion
        "Whether the main circuit is a circom circuit, not a .light file.",
```

File: cli/src/psp-utils/addCircuit.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    ? "psp-template/circuits/circuit_circom"
+    : "psp-template/circuits/circuit_psp";
+
+  await executeCargoGenerate({
+    args: [
+      "generate",
+      // "--git",
+      // "https://github.com/Lightprotocol/psp-template",
+      // --tag,
+      // PSP_TEMPLATE_TAG,
```
Comment (SwenSchaeferjohann) [inline cli/src/psp-utils/addCircuit.ts:28]: can these be removed?
```suggestion
    
```

File: cli/src/psp-utils/buildCircom.ts [hunk 4]
```diff
@@ -17,15 +18,17 @@ import { findFile } from "./utils";
...
+  circuitPath = "./circuits",
 }: {
-  circuitFileName: string;
+  circuitName: string;
   ptau: number;
   programName: string;
+  circuitPath?: string;
 }): Promise<void> {
-  const circuitName = circuitFileName.slice(0, -".circom".length);
+  // const circuitName = circuitFileName.slice(0, -".circom".length);
```
Comment (SwenSchaeferjohann) [inline cli/src/psp-utils/buildCircom.ts:31]: can this be removed?
```suggestion
 
```

File: cli/src/psp-utils/constants.ts [hunk 5]
```diff
@@ -1,3 +1,7 @@
 export const PSP_TEMPLATE_TAG = "v0.1.1";
 export const PROGRAM_TAG = "v0.3.2";
 export const MACRO_CIRCOM_TAG = "v0.1.5";
+export const ZK_JS_VERSION = "file:../light-zk.js"; // "0.3.2-alpha.13";
+export const PROVER_JS_VERSION = "file:../light-prover.js"; // "0.1.0-alpha.1";
```
Comment (SwenSchaeferjohann) [inline cli/src/psp-utils/constants.ts:5]: what do the inline comments indicate? can they be removed? will we have to switch back to relative imports?

Comment (SwenSchaeferjohann) [inline cli/src/psp-utils/constants.ts:5]: what do the inline comments indicate? can they be removed? will we have to switch back to relative imports?
```suggestion
export const ZK_JS_VERSION = "file:../light-zk.js";
export const PROVER_JS_VERSION = "file:../light-prover.js";
```

File: cli/src/psp-utils/createRustVerifyingKey.ts [hunk 6]
```diff
@@ -60,7 +61,7 @@ async function getProofInputsFromSymFile(artifactPath: string) {
 
   let match;
   let keys = [];
-  const symText = fs.readFileSync(`${artifactPath}.sym`, "utf-8");
+  const symText = fs.readFileSync(`${artifactPath}Main.sym`, "utf-8");
   while ((match = regex.exec(symText)) !== null) {
     keys.push(match[1]);
     // const name = match[1];
```
Comment (SwenSchaeferjohann) [inline cli/src/psp-utils/createRustVerifyingKey.ts:67]: can this be removed?
```suggestion

```

File: cli/src/psp-utils/createRustVerifyingKey.ts [hunk 7]
```diff
@@ -105,28 +106,42 @@ async function getProofInputsFromSymFile(artifactPath: string) {
...
     return str.replace(/[A-Z]/g, (letter) => `_${letter.toLowerCase()}`);
   }
 
-  function buildRustType(dimension: number, size: number[]): string {
+  function buildRustType(
+    dimension: number,
+    size: number[],
+    isInstructionData: boolean = false
+  ): string {
+    // the base type is a 254 bit bigint thus a 32 byte array
```
Comment (SwenSchaeferjohann) [inline cli/src/psp-utils/createRustVerifyingKey.ts:122]:
```suggestion
    // the base type is a 254-bit bigint; thus, a 32-byte array
```

File: cli/src/psp-utils/createRustVerifyingKey.ts [hunk 8]
```diff
@@ -105,28 +106,42 @@ async function getProofInputsFromSymFile(artifactPath: string) {
...
   // parse the inputs output into a rust struct as a program account
-  let structDefinition = `\n#[account]\npub struct ZK${circuitName}Inputs {\n`;
+  let structDefinition = `\n#[account]\npub struct ${prefix}${circuitName} {\n`;
 
   preparedInputs.forEach((input) => {
     const { inputName, dimension, size } = input;
-    const rustType = buildRustType(dimension, size);
+    const rustType = buildRustType(dimension, size, isInstructionData);
     const inputName_snake = camelToSnakeCase(inputName);
     structDefinition += `    ${inputName_snake}: ${rustType},\n`;
```
Comment (SwenSchaeferjohann) [inline cli/src/psp-utils/createRustVerifyingKey.ts:146]: are the empty spaces before ${ planned? 
```suggestion
    structDefinition += `${inputName_snake}: ${rustType},\n`;
```

File: cli/src/psp-utils/init.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  var rustName = toSnakeCase(name);
+  let programsType = type === ProjectType.PSP_CIRCOM ? ProjectType.PSP : type;
+
+  await executeCargoGenerate({
+    args: [
+      "generate",
+      // "--git",
+      // "https://github.com/Lightprotocol/psp-template",
+      // --tag,
+      // PSP_TEMPLATE_TAG,
```
Comment (SwenSchaeferjohann) [inline cli/src/psp-utils/init.ts:33]: can this be removed?
```suggestion
    
```

File: cli/src/utils/initTestEnv.ts [hunk 10]
```diff
@@ -27,7 +27,7 @@ export async function initTestEnv({
 
   await start_test_validator({ additonalPrograms, skip_system_accounts });
   const anchorProvider = await setAnchorProvider();
-  await airdropSol({
+  let res = await airdropSol({
```
Comment (SwenSchaeferjohann) [inline cli/src/utils/initTestEnv.ts:30]: is `res` being used?

File: light-zk.js/src/transaction/transaction.ts [hunk 11]
```diff
@@ -215,9 +215,11 @@ export class Transaction {
...
         "compileAndProve",
       );
-    const proofs = [this.getProof()];
-    if (this.appParams) proofs.push(this.getAppProof());
-    await Promise.all(proofs);
+    await this.getProof();
+    if (this.appParams) await this.getAppProof();
+    // const proofs = [this.getProof()];
+    // if (this.appParams) proofs.push(this.getAppProof());
+    // await Promise.all(proofs);
```
Comment (SwenSchaeferjohann) [inline light-zk.js/src/transaction/transaction.ts:222]: can this be removed?
```suggestion
 
```

**Review by SwenSchaeferjohann — APPROVED** [review]
besides minor comments mostly commented code it looks good; did not test locally 
feel free to tag me again when draft->ready

File: cli/src/psp-utils/init.ts (hunk 9 above)
Comment (ananas-block) [reply cli/src/psp-utils/init.ts]: was for testing

File: cli/src/psp-utils/buildCircom.ts (hunk 1 above)
Comment (ananas-block) [reply cli/src/psp-utils/buildCircom.ts]: done

File: cli/src/psp-utils/addCircuit.ts (hunk 3 above)
Comment (ananas-block) [reply cli/src/psp-utils/addCircuit.ts]: was for testing

**Pull request merged**